diff --git a/dist/adapters/node.mjs b/dist/adapters/node.mjs
index ea42cd2f771052f83b48db9cdca5ab71f8136549..af54ed2dda7be2148cece5a6834ae0d4fc3abdc9 100644
--- a/dist/adapters/node.mjs
+++ b/dist/adapters/node.mjs
@@ -207,17 +207,20 @@ const NodeRequest = /* @__PURE__ */ (() => {
 		get _abortController() {
 			if (!this.#abortController) {
 				this.#abortController = new AbortController();
-				const { req, res } = this.runtime.node;
-				const abortController = this.#abortController;
-				const abort = (err) => abortController.abort?.(err);
-				if (res) res.once("close", () => {
-					const reqError = req.errored;
-					if (reqError) abort(reqError);
-					else if (!res.writableEnded) abort();
-				});
-				else req.once("close", () => {
-					if (!req.complete) abort();
-				});
+				// Fix: Guard against undefined runtime.node during Nitro prerendering
+				if (this.runtime?.node) {
+					const { req, res } = this.runtime.node;
+					const abortController = this.#abortController;
+					const abort = (err) => abortController.abort?.(err);
+					if (res) res.once("close", () => {
+						const reqError = req.errored;
+						if (reqError) abort(reqError);
+						else if (!res.writableEnded) abort();
+					});
+					else if (req) req.once("close", () => {
+						if (!req.complete) abort();
+					});
+				}
 			}
 			return this.#abortController;
 		}
